tests/integration/targets/ndo_ipsla_monitoring_policy/tasks/main.yml

```diff
@@ -0,0 +1,518 @@
+      register: nm_delete_ipsla_monitoring_policy
+    
+    - name: Delete an IPSLA monitoring policy again
+      cisco.mso.ndo_ipsla_monitoring_policy:
+        <<: *delete_ipsla_monitoring_policy
+      register: nm_delete_ipsla_monitoring_policy_again
+
+    - name: Assert IPSLA monitoring policy was deleted
+      assert:
+        that:
+          - cm_delete_ipsla_monitoring_policy is changed
+          - cm_delete_ipsla_monitoring_policy.previous.name == 'ansible_test_ipsla_monitoring_policy_changed'
+          - cm_delete_ipsla_monitoring_policy.previous.slaType == 'tcp'
+          - cm_delete_ipsla_monitoring_policy.current == {}
+          - nm_delete_ipsla_monitoring_policy is changed
+          - nm_delete_ipsla_monitoring_policy.previous.name == 'ansible_test_ipsla_monitoring_policy_changed'
+          - nm_delete_ipsla_monitoring_policy.previous.slaType == 'tcp'
+          - nm_delete_ipsla_monitoring_policy.current == {}
+          - nm_delete_ipsla_monitoring_policy_again is not changed
+          - nm_delete_ipsla_monitoring_policy_again.previous == {}
+          - nm_delete_ipsla_monitoring_policy_again.current == {}
+
+    - name: Create another IPSLA monitoring policy
+      cisco.mso.ndo_ipsla_monitoring_policy: &create_another_ipsla_monitoring_policy
+        <<: *mso_info
+        template: ansible_tenant_template
+        ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy2
+        sla_type: http
+        http_version: HTTP11
+        http_uri: /test
+        sla_frequency: 120
+        detect_multiplier: 5
+        req_data_size: 64
+        type_of_service: 16
+        operation_timeout: 1000
+        threshold: 1000
+        ipv6_traffic_class: 32
+        destination_port: 80
+        state: present
+      register: nm_create_another_ipsla_monitoring_policy
+
+    - name: Assert creation of another IPSLA monitoring policy
+      assert:
+        that:
+          - nm_create_another_ipsla_monitoring_policy is changed
+          - nm_create_another_ipsla_monitoring_policy.current.name == "ansible_test_ipsla_monitoring_policy2"
+          - nm_create_another_ipsla_monitoring_policy.current.slaType == "http"
+          - nm_create_another_ipsla_monitoring_policy.current.slaPort == 80
+          - nm_create_another_ipsla_monitoring_policy.current.httpVersion == "HTTP11"
+          - nm_create_another_ipsla_monitoring_policy.current.httpUri == "/test"
+          - nm_create_another_ipsla_monitoring_policy.current.slaFrequency == 120
+          - nm_create_another_ipsla_monitoring_policy.current.slaDetectMultiplier == 5
+          - nm_create_another_ipsla_monitoring_policy.current.reqDataSize == 64
+          - nm_create_another_ipsla_monitoring_policy.current.ipv4ToS == 16
+          - nm_create_another_ipsla_monitoring_policy.current.timeout == 1000
+          - nm_create_another_ipsla_monitoring_policy.current.threshold == 1000
+          - nm_create_another_ipsla_monitoring_policy.current.ipv6TrfClass == 32
```

Comment:
Is this needed? You are already creating a second one for query all, cant you use that one for the uuid delete test or is there another reason for this test?

Reply:
This is mainly for code coverage and not for deletion via uuid. Without this some part of code is not covered under inclusion of attributes within payload when type is http. In the above cases we add update to http from icmp.

Reply:
Perhaps rename the test case to make it clear that this is what is tested?